pkg/datasource/datasource.go

```diff
@@ -263,26 +295,23 @@ func (ds *StravaDatasource) RefreshAccessToken(refreshToken string) (*TokenExcha
 	}, nil
 }
 
-func (ds *StravaDatasource) StravaAPIQuery(ctx context.Context, tsdbReq *datasource.DatasourceRequest) (*datasource.DatasourceResponse, error) {
+func (ds *StravaDatasourceInstance) StravaAPIQuery(ctx context.Context, tsdbReq *StravaAPIRequest) (*StravaApiResourceResponse, error) {
 	accessToken, err := ds.GetAccessToken()
 	if err != nil {
 		return nil, err
 	}
 
-	query := tsdbReq.Queries[0]
-	queryJSON, err := simplejson.NewJson([]byte(query.ModelJson))
-	if err != nil {
-		return nil, err
-	}
-	endpoint := queryJSON.Get("target").Get("endpoint").MustString()
-	params := queryJSON.Get("target").Get("params").MustMap()
+	// TODO: cleanup
+	query := tsdbReq
+	endpoint := query.Endpoint
+	params := query.Params
```

Comment:
Is there a reason why you're not using tsdbReq directly?

Reply:
Hm, probably missed to change this after copying code :)